File: QID-3226-SFEVolaCov/SFEVolaCov.R

# clear variables and close windows
rm(list = ls(all = TRUE))
graphics.off()

# read data
x = read.table("implvola.dat")

# rescale
x = x/100

# compute first differences
z = apply(x,2,diff)
# calculate covariance
s = cov(z) * 1e+05
round(s, 2)

File: QID-3226-SFEVolaCov/SFEVolaCov.m
clear
clc
close all

%Load data
x = load('implvola.dat');

x = x / 100;
n = length(x);

%Calculate first differences and calculate the covariance matrix.
z = x(2:n, :) - x(1:(n - 1), :);
s = cov(z) * 100000;

disp('Empirical Covariance Matrix')
disp('      Sub 1    Sub 2     Sub 3     Sub 4     Sub 5     Sub 6     Sub 7     Sub 8')
disp(s)